safehttp/plugins/cors/cors_test.go

```diff
@@ -283,24 +271,18 @@ func TestRequestDisallowedContentTypes(t *testing.T) {
 				req.Header.Set("Content-Type", ct)
 			}
 
-			rr := safehttptest.NewResponseRecorder()
+			fakeRW, rr := safehttptest.NewFakeResponseWriter()
 
 			it := cors.Default("https://foo.com")
-			it.Before(rr.ResponseWriter, req, nil)
+			it.Before(fakeRW, req, nil)
 
-			if want := safehttp.StatusUnsupportedMediaType; rr.Status() != want {
-				t.Errorf("rr.Status() got: %v want: %v", rr.Status(), want)
-			}
-			wantHeaders := map[string][]string{
-				"Content-Type":           {"text/plain; charset=utf-8"},
-				"X-Content-Type-Options": {"nosniff"},
+			if want := safehttp.StatusUnsupportedMediaType; rr.Code != int(want) {
+				t.Errorf("rr.Code got: %v want: %v", rr.Code, want)
 			}
```

Comment:
I've dropped the headers/body checks in many cases where it felt like a change detector.

Most importantly, headers and bodies in these tests were coming from the `Dispatcher` implementation, not the `Interceptor` that was under test, so they don't really help testing the interceptor's logic.